File: VB/11_SecurityAndSignatures/AddInvisibleSignature.vb
Imports Spire.Pdf
Imports Spire.Pdf.Interactive.DigitalSignatures
Imports System.Security.Cryptography.X509Certificates

Namespace AddInvisibleSignature
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Specify the PFX certificate file path
			Dim pfxPath As String = "..\..\..\..\..\..\Data\gary.pfx"

			' Specify the input PDF file path
			Dim inputPdf As String = "..\..\..\..\..\..\Data\AddInvisibleSignature.pdf"

			' Specify the output PDF file path
			Dim outputPdf As String = "AddInvisibleSignature_result.pdf"

			' Create a new instance of PdfDocument
			Dim pdf As New PdfDocument()

			' Load the PDF document from the specified input file path
			pdf.LoadFromFile(inputPdf)

			' Create a new X509Certificate2 object using the PFX certificate file and password
			Dim x509 As New X509Certificate2(pfxPath, "e-iceblue")

			' Create a PdfOrdinarySignatureMaker object for the document with the X509 certificate
			Dim signatureMaker As New PdfOrdinarySignatureMaker(pdf, x509)

			' Make an invisible signature in the PDF document with the specified field name
			signatureMaker.MakeSignature("signName")

			' Save the modified PDF document to the specified output file path
			pdf.SaveToFile(outputPdf, FileFormat.PDF)

			' Close the PDF document
			pdf.Close()

			' Launch the pdf file
			FileViewer(outputPdf)
		End Sub
		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/11_SecurityAndSignatures/AddInvisibleSignature.cs

using Spire.Pdf;
using Spire.Pdf.Interactive.DigitalSignatures;
using System;
using System.Security.Cryptography.X509Certificates;
using System.Windows.Forms;

namespace AddInvisibleSignature
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Input, output, and certificate file paths.
            string pfxPath = @"..\..\..\..\..\..\Data\gary.pfx";
            string inputPdf = @"..\..\..\..\..\..\Data\AddInvisibleSignature.pdf";
            string outputPdf = "AddInvisibleSignature_result.pdf";

            // Create a new PDF document.
            PdfDocument pdf = new PdfDocument();

            // Load the input PDF file from the disk.
            pdf.LoadFromFile(inputPdf);

            // Create an X509Certificate2 object by loading the certificate file with the specified password.
            X509Certificate2 x509 = new X509Certificate2(pfxPath, "e-iceblue");

            // Create an instance of PdfOrdinarySignatureMaker using the loaded document and certificate.
            PdfOrdinarySignatureMaker signatureMaker = new PdfOrdinarySignatureMaker(pdf, x509);

            // Create an invisible signature with the specified name.
            signatureMaker.MakeSignature("signName");

            // Save the modified PDF document to the specified output file.
            pdf.SaveToFile(outputPdf, FileFormat.PDF);

            //Open the pdf file
            FileViewer(outputPdf);
        }
        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
